### PR TITLE
extensible batch processing plugin based on packaged paws workflows

### PR DESCRIPTION
**Pull request by lensonp**
Hi Ron,

This PR is for revisions on the paws-based batch processing plugin. These changes reflect what I am going to write about paws in the Xi-cam paper, so please take a look and let me know if you think big changes are needed.

1. many refinements to the paws api had to be rolled into the plugin. the version requirement is now >0.8.4. I added a requirements.txt file with this and many other requirements. not sure how you meant to handle requirements- happy to do this differently if you say so.

2. smoothed the interface between the plugin and paws. mostly, I was able to make the plugin extensible to other workflows and remove many lines from it by adding datatype specifications to the paws Operations, and then using these datatypes in the ParameterTree setup.

3. made the plugin extensible. there is now a "XICAM" sub-package of workflows in paws (as of v0.8.4), which are curated specifically for use in Xi-cam batch processing (i separated them to ensure that any future developer who changes them will be aware of this dependency). to show that the plugin is extensible, I included two workflows in it. one, batch integration (same as before but with more features), and two, batch guinier-porod spectrum fitting. any number of other workflows can be handled by adding just a few lines to the plugin. i added a combobox to the gui for workflow selection.

4. .gitignore swap files

There are also two small gui-related details that I didn't finish yet (I will add them to the PR before it gets merged, but wanted to get this on your desk tonight). 

1. the "Run" button will change to "Stop" after the workflow is running, and clicking "Stop" will stop the workflow. 

2. the visualizer will be made to update automatically as the operations execute (currently it only updates when the corresponding item is clicked). i think this is somewhat optional, and maybe deserves a "realtime visualization" toggle, since widget creation can be kind of heavy.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+pyside
```
Comment (ronpandolfi) [inline requirements.txt]: Xi-cam uses setup.py for the requirements; see `install_requires = [...]`. Lets merge these into there. It looks like you've added the following dependencies: 

> sklearn, six, matplotlib, silx, pypaws>0.8.4, pypif, pytz, tzlocal, saxskit

Are these required for the batch plugin and paws?

Comment (ronpandolfi) [reply requirements.txt]: Also, should it be `pypaws>=0.8.4`? The latest I see in pip is 0.8.4.

Comment (lensonp) [reply requirements.txt]: Yes- your'e right about the version.

As for the long comment in __init__, I actually didn't mean for that comment to be a docstring. 

I messed up those dependencies a bit (copy pasted them from elsewhere)- we actually don't need most of those.

I'll push a commit shortly for all of the above.

**Comment by lensonp** [comment]
the only requirements actually needed were:
pypaws>=0.8.4
matplotlib

The matplotlib dependency is for generating plots on systems that can't support pyqtgraph (e.g. rhel6). Not needed for Xi-cam, but it ends up being imported when using paws gui modules. I can probably find a way to get rid of that dependency.

**Comment by lensonp** [comment]
I forgot one. saxskit is a dependency for the guinier-porod saxs spectrum fitting workflow... and saxskit depends on scikit-learn.

**Pull request merged**